src/integrationtests/java/util/JenkinsManager.java

```diff
@@ -0,0 +1,213 @@
+package util;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.apache.http.client.ClientProtocolException;
+import org.json.JSONObject;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+
+import com.ericsson.ei.utils.HttpRequest;
+import com.ericsson.ei.utils.HttpRequest.HttpMethod;
+
+public class JenkinsManager {
+
+    private static final String JENKINS_JOB_TEMPLATE_FILE_PATH = String.join(File.separator, "src", "functionaltests",
+            "resources", "jenkinsJobTemplate.xml");
+    private static final String DEFAULT_JOB_TOKEN = "test";
+    private static final String DEFAULT_SCRIPT = "echo &quot;Test&quot;";
+    private static final String DEFAULT_JOB_NAME = "test_job";
+
+    private String host;
+    private int port;
+    private String encoding;
+    private String crumb;
+
+    /**
+     * Constructor, takes jenkins host, port, username and password.
+     * @param host
+     * @param port
+     * @param username
+     * @param password   **  Jenkins password or API-token
+     * @throws UnsupportedEncodingException
+     * @throws URISyntaxException
+     */
+    public JenkinsManager(String host, int port, String username, String password)
+            throws UnsupportedEncodingException, URISyntaxException {
+        String authString = String.join(":", username, password);
+
+        this.host = host;
+        this.port = port;
+        this.encoding = DatatypeConverter.printBase64Binary(authString.getBytes("utf-8"));
+        this.crumb = getCrumb();
+    }
+
+    /**
+     * Takes a token and script and returns XML data for jenkins with job_token and
+     * script added.
+     *
+     * @param job_token
+     * @param script
+     * @return
+     * @throws IOException
+     */
+    public String getXmlJobData(String job_token, String script) throws IOException {
+        String jobData = getStringFromFile(JENKINS_JOB_TEMPLATE_FILE_PATH);
+
+        if (job_token == null) {
+            job_token = DEFAULT_JOB_TOKEN;
+        }
+        if (script == null) {
+            script = DEFAULT_SCRIPT;
+        }
+
+        jobData = injectScriptAndTokenIntoXmlString(jobData, job_token, script);
+        return jobData;
+    }
+
+    /**
+     * Creates a jenkins job with a given name using the XML data as input for
+     * job configuration
+     *
+     * @param jobName
+     * @param jobXmlData
+     * @return
+     * @throws URISyntaxException
+     * @throws ClientProtocolException
+     * @throws IOException
+     */
+    public boolean createJob(String jobName, String jobXmlData) throws URISyntaxException {
```

Comment:
Don't we have a General HTTPRequest class already in the Bridge functionality classes that can be used for these requests?

Reply:
I use the general HTTPRequest class existing in functional tests.

Reply:
import com.ericsson.ei.utils.HttpRequest;
import com.ericsson.ei.utils.HttpRequest.HttpMethod;

Reply:
Ok.